packages/flutter/test/widgets/text_test.dart

```diff
@@ -1070,6 +1070,7 @@ void main() {
     final ui.Paragraph paragraph = builder.build();
     paragraph.layout(const ui.ParagraphConstraints(width: 1000));
     expect(paragraph.getBoxesForRange(2, 2), isEmpty);
+    paragraph.dispose();
```

Comment:
`WidgetTester` performs checks for dangling app state, such as mouse pointers. Should we also have it proactively look for leaked paragraph?

Or perhaps the leak detector that @polina-c is working on can help here?

Reply:
We'd need changes in dart:ui to support this.

Polina's work should help here. There's another option where we make the engine itself aware of this kind of thing and throw an exception when it happens, but that's possibly a heavy hammer and prone to flakes.